io/pci/pci_info_lscfg.py

```diff
@@ -0,0 +1,172 @@
+        Compare Physical slot info captured from lscfg and lspci
+
+        :param phy_slot_lscfg: Physical slot captuted from lscfg
+        :param phy_slot_lspci: Physical slot captured from lspci
+
+        :Return: Boolean value
+                 True: If both values are false
+                 False: If both values are not same
+        """
+        phy_slot_lscfg = phy_slot_lscfg[:phy_slot_lscfg.rfind('-')]
+        self.log.info("Physical Slot from lscfg: %s", phy_slot_lscfg)
+        self.log.info("Physical Slot from lspci: %s", phy_slot_lspci)
+        if phy_slot_lscfg == phy_slot_lspci:
+            return True
+        return False
+
+    def compare_pci_slot_id_lscfg_lspci(self,
+                                        pci_slot_lscfg,
+                                        pci_slot_lspci):
+        """
+        Compare PCI slot ID info captured from lscfg and lspci
+
+        :param pci_slot_lscfg: pci slot captuted from lscfg
+        :param pci_slot_lspci: pci slot captured from lspci
+
+        :Return: Boolean value
+                 True: If both values are false
+                 False: If both values are not same
+        """
+        self.log.info("PCI Slot from lscfg: %s", pci_slot_lscfg)
+        self.log.info("PCI Slot from lspci: %s", pci_slot_lspci)
+        if pci_slot_lspci == pci_slot_lscfg:
+            return True
+        return False
+
+    def compare_device_id_lspci_lscfg(self,
+                                      device_id_lscfg,
+                                      device_id_lspci):
+        """
+        Compare Device ID info captured from lscfg and lspci
+
+        :param device_id_lscfg: Device ID captuted from lscfg
+        :param device_id_lspci: Device ID captured from lspci
+
+        :Return: Boolean value
+                 True: If both values are false
+                 False: If both values are not same
+        """
+        device_id_lscfg = device_id_lscfg[4:]
+        self.log.info("Device ID from lscfg: %s", device_id_lscfg)
+        self.log.info("Device ID from lspci: %s", device_id_lspci)
+        if device_id_lscfg == device_id_lspci:
+            return True
+        return False
+
+    def compare_sub_vendor_lspci_lscfg(self,
+                                       sub_vendor_lscfg,
+                                       sub_vendor_lspci):
+        """
+        Compare Device ID info captured from lscfg and lspci
+
+        :param sub_vendor_lscfg: Sub vendor captuted from lscfg
+        :param sub_vendor_lspci: Sub vendor captured from lspci
+
+        :Return: Boolean value
+                 True: If both values are false
+                 False: If both values are not same
+        """
+        sub_vendor_lscfg = sub_vendor_lscfg[0:4]
+        self.log.info("Sub vendor ID from lscfg: %s", sub_vendor_lscfg)
+        self.log.info("Sub vendor ID from lspci: %s", sub_vendor_lspci)
+        if sub_vendor_lscfg == sub_vendor_lspci:
+            return True
+        return False
```

Comment:
you can merge all the compare functions into a generic method which takes 2 generic arguments .. so you can avoid many redundant methods.. here is how

```
def   compare_lspci_lscfg(self, val_cfg, val_pci):
     if val_lscfg == val_pci:
                return True
     return False
```
.
.
.
and when calling the same function mutliple places like this
```
if(self.compare_lscfg_lspci(cfg_output['slot'],
                   pci_information_dict['PhySlot'])):

```
```
 if(self.compare_lspci_lscfg(
                   cfg_output['subvendor_device'],
                   pci_information_dict['SVendor'])):
```